daemon/CryptoImpl/cryptorequestprocessor.cpp

```diff
@@ -29,6 +29,33 @@ namespace {
         }
         return retn;
     }
+
+    void nullifyKeyFields(Sailfish::Crypto::Key *key, Sailfish::Crypto::StoredKeyRequest::KeyComponents keep) {
```

Comment:
I am not sure I understand why the `nullifyKeyFields` exists more than once in the code. My thought that it would work like this:

1. We create a crypto request and send it.
2. The daemon receives it and asks the appropriate plugin to get the key.
3. The plugin returns the key data.
4. The code in the crypto request processor calls `nullifyKeyFields`.

But I see this function three times in this PR. Did I miss the conception?

Reply:
The reason is that, in the case where the crypto plugin is also a storage plugin, we never want the secret key data to leak back to the daemon address space, in the case where the client only requests PublicKeyData.  Thus, the plugin (and not the daemon) should implement the field nullification.  While in our case the plugin is loaded in-process it probably doesn't make much difference, but it certainly does for TEE/TPM case.

The "fallback" case (where the daemon performs the nullification) is only applied where the key is stored as a "normal" secret in a secrets storage plugin (rather than crypto storage plugin).

However, the third case (opensslcryptoplugin) should NOT have that method, since it is not a storage plugin.  I copy pasted it there incorrectly, and I will remove that copy of it.

Reply:
@chriadam Good, thanks for the explanation.